### PR TITLE
Upd supervisor

### PR DESCRIPTION
**Pull request by mereacre**

### CODE REVIEW

**Comment by aloisklink** [comment]
Hi @mereacre, I'm going to wait to review this until #154 is merged, most of the changes in this are in that too, if that's okay with you.

**Review by aloisklink — COMMENTED** [review]
Looks good! Thanks for renaming `domain` to `sockctl`, it actually makes it a lot clearer. `sockctl` is a much better name for it.

File: src/supervisor/cmd_processor.h [hunk 1]
```diff
@@ -73,8 +73,10 @@
 #define CMD_SIGN_BLOB "SIGN_BLOB"
 #endif
 
-#define OK_REPLY "OK"
-#define FAIL_REPLY "FAIL"
+#define CMD_DELIMITER 0x20
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/supervisor/cmd_processor.h:76]: Ok

File: src/supervisor/subscriber_events.c [hunk 2]
```diff
@@ -30,34 +30,55 @@
...
+  if (a->type != b->type) {
+    return -1;
+  } else {
+    if (a->type == SOCKET_TYPE_DOMAIN) {
+      return os_memcmp(a->addr_un.sun_path, b->addr_un.sun_path, a->len);
+    } else if (a->type == SOCKET_TYPE_UDP) {
+      if (a->addr_in.sin_port != b->addr_in.sin_port) {
+        return (a->addr_in.sin_port < b->addr_in.sin_port)
+                   ? -1
+                   : (a->addr_in.sin_port > b->addr_in.sin_port);
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/supervisor/subscriber_events.c:50]: Missed that one :+1:

File: src/supervisor/supervisor.c [hunk 3]
```diff
@@ -274,58 +274,83 @@ void ap_service_callback(struct supervisor_context *context, uint8_t mac_addr[],
...
+}
+
+void eloop_read_domain_handler(int sock, void *eloop_ctx, void *sock_ctx) {
+  (void)eloop_ctx;
+
+  struct client_address claddr;
+  struct supervisor_context *context = (struct supervisor_context *)sock_ctx;
+
+  os_memset(&claddr, 0, sizeof(struct client_address));
+  claddr.type = SOCKET_TYPE_DOMAIN;
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/supervisor/supervisor.c:335]: Thanks for pointing it out.

File: src/utils/sockctl.h [hunk 4]
```diff
@@ -18,30 +18,33 @@
...
+};
 
 /**
  * @brief Client address structure definition
  *
  */
 struct client_address {
-  struct sockaddr_un addr;
+  struct sockaddr_un addr_un;
+  struct sockaddr_in addr_in;
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/utils/sockctl.h:45]: Nice catch. I replaced in all places.

File: src/supervisor/supervisor.c [hunk 5]
```diff
@@ -274,58 +274,83 @@ void ap_service_callback(struct supervisor_context *context, uint8_t mac_addr[],
...
+    os_free(buf);
+    utarray_free(args);
+    return -1;
   }
 
-  ptr = (char **)utarray_next(cmd_arr, ptr);
+  os_free(buf);
+
+  char **arg = NULL;
+  arg = (char **)utarray_next(args, arg);
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/supervisor/supervisor.c:313]: Thanks

File: src/utils/net.h [hunk 6]
```diff
@@ -108,4 +137,23 @@ uint8_t get_short_subnet(char *subnet_mask);
...
+ * @brief Disable the PMTU discovery for sockets
+ *
+ * @param[in] sock The socket descriptor
+ * @return 0 on success, -1 on failure
+ */
+int disable_pmtu_discovery(int sock);
+
+/**
+ * @brief Convert ASCII string to MAC address (in any known format)
+ *
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/utils/net.h:151]: Thanks

File: src/utils/os.h [hunk 7]
```diff
@@ -241,6 +196,14 @@ int os_get_random_number_s(unsigned char *buf, size_t len);
  */
 int hex2byte(const char *hex);
 
+/**
+ * @brief Hex char two number
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/utils/os.h:200]: Thanks

**Pull request merged**